### DISCUSSION TITLE
Enable or disable camera in raspi-config

### DISCUSSION THREAD

**Discussion by TurboTronix** [mainsail-crew/spyglass#91] [General]
Do we need to enable to disable the camera from raspi-config
![image](https://github.com/user-attachments/assets/9cfbc6d7-3c7c-435c-8d2c-faf169dec41c)

**Comment by mryel00** [comment]
You don't enable/disable the cam like that. That's legacy camera support. For Spyglass you should not activate legacy support.

**Reply by TurboTronix** [reply]
Yes I got an error saying to disable when I run.py

**Reply by mryel00** [reply]
I recommend to send such errors, as I cannot read your screen.

**Reply by TurboTronix** [reply]
```
INFO:spyglass:Spyglass 0.15.0
[0:01:08.317510286] [1628]  INFO Camera camera_manager.cpp:297 libcamera v0.0.5+                         83-bde9b04f
ERROR:picamera2.picamera2:Camera(s) not found (Do not forget to disable legacy c                         amera with raspi-config).
ERROR:picamera2.picamera2:Camera __init__ sequence did not complete.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/picamera2/picamera2.py", line 242, in __i                         nit__
    self._open_camera()
  File "/usr/lib/python3/dist-packages/picamera2/picamera2.py", line 442, in _op                         en_camera
    if not self._initialize_camera():
  File "/usr/lib/python3/dist-packages/picamera2/picamera2.py", line 406, in _in                         itialize_camera
    raise RuntimeError("Camera(s) not found (Do not forget to disable legacy cam                         era with raspi-config).")
RuntimeError: Camera(s) not found (Do not forget to disable legacy camera with r                         aspi-config).

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pi/spyglass/./run.py", line 6, in <module>
    raise SystemExit(main())
  File "/home/pi/spyglass/spyglass/cli.py", line 48, in main
    cam = init_camera(
  File "/home/pi/spyglass/spyglass/camera/__init__.py", line 20, in init_camera
    picam2 = Picamera2(camera_num, tuning=tuning)
  File "/usr/lib/python3/dist-packages/picamera2/picamera2.py", line 254, in __i                         nit__
    raise RuntimeError("Camera __init__ sequence did not complete.")
RuntimeError: Camera __init__ sequence did not complete.

```

**Reply by mryel00** [reply]
I think it's maybe saying, to disable legacy stack? Then you should disable it ofc
(I sent it before I saw your message)

**Reply by TurboTronix** [reply]
I disabled it but still can't find my camera

**Reply by mryel00** [reply]
Then check that you inserted the ribbon correctly. `libcamera-hello --list-cameras` should list all the cameras available. If it cannot find anything, try a restart and if that doesn't help it's most likely attached wrong.

**Reply by TurboTronix** [reply]
Yeah something is wrong with my camera...

```
pi@mainsailos:~/spyglass $ libcamera-hello --list-cameras
No cameras available!
```

**Reply by TurboTronix** [reply]
tried numerous cables, guess the caemra is dead

**Reply by mryel00** [reply]
Send a picture of the port that you connected the cam to on your Pi

**Reply by TurboTronix** [reply]
![PXL_20240924_130817310.jpg](https://github.com/user-attachments/assets/2f095b1b-f984-473c-b02b-edc25cd1fa24)

![PXL_20240924_130810041.jpg](https://github.com/user-attachments/assets/476efcb7-350f-46c9-8902-ac69cd711a2a)

![PXL_20240924_130801907.jpg](https://github.com/user-attachments/assets/4965c986-8241-4e1c-98ad-c4e6ae989809)

**Reply by TurboTronix** [reply]
When i was taking pictures i noticed the clip for the lens sensor was off, that fixed it :)

Now it works
